src/Data/List/Fresh/NonEmpty.agda

```diff
@@ -0,0 +1,47 @@
+{-# OPTIONS --without-K --safe #-}
+
+module Data.List.Fresh.NonEmpty where
+
+open import Level using (Level; _⊔_)
+open import Data.List.Fresh as List# using (List#; []; cons; fresh)
+open import Data.Maybe.Base using (Maybe; nothing; just)
+open import Data.Nat using (ℕ; suc)
+open import Data.Product using (_×_; _,_)
+open import Data.Unit.Polymorphic.Base using (⊤; tt)
+open import Relation.Binary as B using (Rel)
+
+private
+  variable
+    a r : Level
+    A : Set a
+
+module _ {a} (A : Set a) (R : Rel A r) where 
+
+  record List#⁺ : Set (a ⊔ r) where 
+    field
+      head : A
+      tail : List# A R
+      rel : fresh A R head tail
+
+    infixr 5 _∷#_
+    pattern _∷#_ x xs = record { head = x; tail = xs }
```

Comment:
Unfortunately this is likely to clash with the `List#` one because we
do not have (yet) type-directed disambiguation of pattern synonyms.
You could maybe instead have:

```suggestion
  infixr 5 _∷#⁺_
  record List#⁺ : Set (a ⊔ r) where
    constructor _∷#⁺_
    field
      head  : A
      tail  : List# A R
      {rel} : fresh A R head tail
```
?